webapp/packages/plugin-navigation-tree/src/NavigationTree/ElementsTree/useElementsTree.ts

```diff
@@ -170,98 +169,96 @@ export function useElementsTree(options: IOptions): IElementsTree {
       return false;
     }
   }
+  const [loadingNodes] = useState(() => observable.map(new Map<string, Promise<void>>(), { deep: false }));
 
   const functionsRef = useObjectRef({
-    async loadTree(nodeId: string) {
-      elementsTree.loading = true;
-      try {
-        await projectInfoResource.load();
-        await connectionInfoResource.load(ConnectionInfoActiveProjectKey);
-        const preloadedRoot = await elementsTree.loadPath(options.folderExplorer.state.fullPath);
-
-        if (preloadedRoot !== options.folderExplorer.state.folder) {
-          if (preloadedRoot === undefined) {
-            options.folderExplorer.open([], options.baseRoot);
-          } else {
-            this.exitNodeFolder(preloadedRoot);
-          }
-          return;
+    async loadTree(...nodes: string[]) {
+      await Promise.all(loadingNodes.values());
```

Comment:
I suppose we can move it a little bit down in order to preload all resources first and when all data  is being fresh we can start working with upcoming tree nodes